pkg/streamingpromql/operators/aggregations/aggregation.go

```diff
@@ -125,6 +125,12 @@ func (a *Aggregation) SeriesMetadata(ctx context.Context) ([]types.SeriesMetadat
 	}
 
 	defer types.SeriesMetadataSlicePool.Put(&innerSeries, a.MemoryConsumptionTracker)
+	// TODO: place this as part of Put call above.
+	defer func() {
+		for _, s := range innerSeries {
+			a.MemoryConsumptionTracker.DecreaseMemoryConsumptionForLabels(s.Labels)
+		}
+	}()
```

Comment:
Is this still needed? Looks like this should be handled by the `Put` hook now.

Reply:
Remnant from rebase process. I will remove that.